### PR TITLE
Type fixes needed to make matrix-appservice-irc build

### PR DESCRIPTION
**Pull request by Half-Shot**
This is a collection of changes needed to make our types work with the IRC bridge. We should either merge this into the parent PR (and subsequently into that one), or find a way to include these changes.

### CODE REVIEW

File: src/components/state-lookup.ts [hunk 1]
```diff
@@ -33,14 +33,15 @@ interface StateLookupRoom {
...
-interface StateLookupEvent {
+export interface StateLookupEvent {
     // eslint-disable-next-line camelcase
     room_id: string;
     // eslint-disable-next-line camelcase
     state_key: string;
     type: string;
     // eslint-disable-next-line camelcase
     event_id: string;
+    content: unknown;
```
Comment (jaller94) [inline src/components/state-lookup.ts:44]: Is content guaranteed to be an `Object` or claimed as here `unknown`? That would require some more type checks in my PR.

Comment (Half-Shot) [reply src/components/state-lookup.ts:44]: We could probably call it a `Record<string,unknown>`

Comment (jaller94) [reply src/components/state-lookup.ts:44]: But if that's not guaranteed by the client lib, we'll get errors like the one #155 fixes, where a presumed object may be a string or a number.

Comment (Half-Shot) [reply src/components/state-lookup.ts:44]: Content should really never be anything other than a object, if it is, we should reject it before insertion?

Comment (Half-Shot) [reply src/components/state-lookup.ts:44]: https://github.com/matrix-org/matrix-appservice-bridge/pull/196/commits/e027f071835160d106616950564bf8d787d3531f

File: src/components/state-lookup.ts [hunk 2]
```diff
@@ -191,9 +187,22 @@ export class StateLookup {
...
-            r.events[event.type] = {};
+        this.insertEvent(r, event);
+    }
+
+    private insertEvent(roomSet: StateLookupRoom, event: StateLookupEvent) {
+        if (typeof event.content !== "object") {
+            // Reject - unexpected content type
+            return;
+        }
+        if (!event.type || !event.state_key) {
```
Comment (jaller94) [inline src/components/state-lookup.ts:198]: Just for learning something: This broke the tests. It's not testing if these keys are missing (as claimed by the line below), but if they are truethy. The tests use empty strings, which aren't truethy.

Comment (jaller94) [reply src/components/state-lookup.ts:198]: Fixed in https://github.com/matrix-org/matrix-appservice-bridge/pull/185/commits/deb2f38ab33b74567bba6be879440da1d26f5a4b

**Pull request merged**